pkg/sql/sqlbase/validate_test.go

```diff
@@ -0,0 +1,238 @@
+			"EncodingType":      {status: thisFieldReferencesNoObjects},
+			"Sharded":           {status: iSolemnlySwearThisFieldIsValidated},
+			"Disabled":          {status: thisFieldReferencesNoObjects},
+			"GeoConfig":         {status: thisFieldReferencesNoObjects},
+			"Predicate": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(mgartner): add validation"},
+		},
+	},
+	{
+		obj: ColumnDescriptor{},
+		fieldMap: map[string]validationStatusInfo{
+			"Name": {status: thisFieldReferencesNoObjects},
+			"ID":   {status: thisFieldReferencesNoObjects},
+			"Type": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(rohan): add validation for custom types"},
+			"Nullable": {status: thisFieldReferencesNoObjects},
+			"DefaultExpr": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(features): add validation"},
+			"Hidden": {status: thisFieldReferencesNoObjects},
+			"UsesSequenceIds": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(features): add validation"},
+			"OwnsSequenceIds": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(features): add validation"},
+			"ComputeExpr": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(features): add validation"},
+			"LogicalColumnID": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(features): add validation"},
+			"AlterColumnTypeInProgress": {status: thisFieldReferencesNoObjects},
+		},
+	},
+	{
+		obj: ForeignKeyConstraint{},
+		fieldMap: map[string]validationStatusInfo{
+			"OriginTableID": {status: iSolemnlySwearThisFieldIsValidated},
+			"OriginColumnIDs": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(schema): add validation"},
+			"ReferencedColumnIDs": {
+				status: todoIAmKnowinglyAddingTechDebt,
+				reason: "initial import: TODO(schema): add validation"},
+			"ReferencedTableID": {status: iSolemnlySwearThisFieldIsValidated},
+			"Name":              {status: thisFieldReferencesNoObjects},
+			"Validity":          {status: thisFieldReferencesNoObjects},
+			"OnDelete":          {status: thisFieldReferencesNoObjects},
+			"OnUpdate":          {status: thisFieldReferencesNoObjects},
+			"Match":             {status: thisFieldReferencesNoObjects},
+		},
+	},
+}
+
+type validationStatusInfo struct {
+	status validateStatus
+	reason string
+}
+
+// Hello! If you're seeing this test fail, you probably just added a new
+// protobuf field to a descriptor. Please read the documentation at the top of
+// this file.
+func TestValidateCoversAllDescriptorFields(t *testing.T) {
+	for _, structInfo := range validationMap {
+		o := structInfo.obj
+		for field, info := range structInfo.fieldMap {
+			switch info.status {
+			case statusUnvalidated:
+				t.Logf("field %T.%s marked as unvalidated", o, field)
```

Comment:
Any reason why these aren't `t.Error`'s?

Reply:
No, I forgot about that API. done.